include/babeltrace/debuginfo-wrapper.h

```diff
@@ -0,0 +1,130 @@
+#ifndef _BABELTRACE_DEBUGINFO_WRAPPER_H
+#define _BABELTRACE_DEBUGINFO_WRAPPER_H
+
+#include <babeltrace/ctf-ir/metadata.h>
+
+#ifdef ENABLE_DEBUGINFO
+
+#include <babeltrace/debuginfo.h>
+#include <babeltrace/ctf-text/types.h>
+
+static inline
+void ctf_text_integer_write_debug_info(struct bt_stream_pos *ppos,
+				       struct bt_definition *definition)
+{
+	struct definition_integer *integer_definition =
+		container_of(definition, struct definition_integer, p);
+	struct ctf_text_stream_pos *pos = ctf_text_pos(ppos);
+	struct debug_info_source *debug_info_src =
+		integer_definition->debug_info_src;
+
+	/* Print debug info if available */
+	if (debug_info_src) {
+		if (debug_info_src->func || debug_info_src->filename) {
+			int add_comma = 0;
+
+			fprintf(pos->fp, ", debug_info = { ");
+
+			if (debug_info_src->func) {
+				fprintf(pos->fp, "func = \"%s\"", debug_info_src->func);
+				add_comma = 1;
+			}
+
+			if (debug_info_src->filename) {
+				if (add_comma) {
+					fprintf(pos->fp, ", ");
+				}
+
+				fprintf(pos->fp, "source_loc = \"%s:%lu\"",
+					debug_info_src->filename,
+					debug_info_src->line_no);
+			}
+
+			fprintf(pos->fp, " }");
+		}
+	}
+}
+
+static inline
+int trace_debug_info_create(struct ctf_trace *trace)
+{
+	int ret = 0;
+
+	if (strcmp(trace->env.domain, "ust") != 0) {
+		goto end;
+	}
+
+	if (strcmp(trace->env.tracer_name, "lttng-ust") != 0) {
+		goto end;
+	}
+
+	trace->debug_info = debug_info_create();
+
+	if (!trace->debug_info) {
+		ret = -1;
+		goto end;
+	}
+
+end:
+	return ret;
+}
+
+static inline
+void trace_debug_info_destroy(struct ctf_trace *trace)
+{
+	debug_info_destroy(trace->debug_info);
+}
+
+static inline
+void ctf_init_debug_info()
+{
+	int ret;
+
+	ret = debug_info_init();
+	assert(!ret);
```

Comment:
No reason, I just moved this here from the `ctf.c` constructor where it initially asserted, I'll just return the value here and `return 0` in the default/disabled implementation instead, and assert there.